### ISSUE TITLE
Ability to Create General Purpose Serverless Sql Azure Databases

### ISSUE THREAD

**Issue by groogiam** [Azure/azure-libraries-for-net#1229]
**Is your feature request related to a problem? Please describe.**
Feature Request (I think). The api does not appear to support server less azure sql resources. ServiceObjectName class lacks GP values.

**Describe the solution you'd like**
I would like the ability to create a General Purpose serverless compute as defined here https://docs.microsoft.com/en-us/sql/t-sql/statements/create-database-transact-sql?view=azuresqldb-current&tabs=sqlpool

E.g. GP_S_Gen5_4

**Describe alternatives you've considered**

**Additional context**
None

**Information Checklist**
Kindly make sure that you have added all the following information above and checkoff the required fields otherwise we will treat the issuer as an incomplete report
- [x] Description Added
- [x] Expected solution specified

**Comment by weidongxu-microsoft** [comment]
For .NET, the development focus has shifted to the next generation of Azure SDKs which follows the new SDK guideline and introduces a set of important new features. Those new packages are currently in preview state and we are actively working on making it production ready. You can visit this link here to see the latest .NET packages:
https://devblogs.microsoft.com/azure-sdk/october-2020-management-ga/

We have also published a few blog posts on why we are doing this:
https://devblogs.microsoft.com/azure-sdk/introducing-new-previews-for-azure-management-libraries/

With this background, .NET Fluent is currently in a low maintenance mode, and we mostly do security and bug fixes. It is subject to deprecation in the future when the new set of .NET packages become Generally Available (GA). Please let us know if there are further questions, thanks!

**Comment by groogiam** [comment]
Thanks for the info. From the links provided it does not look like there .NET Management libraries for SQL or AppService in this format yet. The post only seems to reference 

.NET
Cosmos DB
DNS
Insight (Monitor)

Am I missing something?

**Comment by weidongxu-microsoft** [comment]
@groogiam 

Yes, they are not there yet. The latest can be queried here https://azure.github.io/azure-sdk/releases/latest/mgmt/dotnet.html

From what I heard, the next phase of the new .NET SDK is previewing new feature on June. GA is expected to be within 2021.

**Comment by groogiam** [comment]
@weidongxu-microsoft Any info on how long it will be before the fluent packages are officially deprecated? It sounds like I have no choice right now to use those packages. I just want to understand how much runway I have until they are officially not supported.

**Comment by weidongxu-microsoft** [comment]
@nickzhums for schedule on new and old .NET SDK.

We do not have official deprecation schedule yet.

Another alternative (with more frequent update, but likely less stable API) is here:
https://www.nuget.org/packages/Microsoft.Azure.Management.Sql

Note that this lib is to be replaced as well.

**Comment by Tyler-V** [comment]
Looks like this may not have been resolved but I am looking for the same option,

Is there a way to provision this using the azure management sql fluent API?

![image](https://user-images.githubusercontent.com/3893234/174116884-051955e8-7f36-4cda-b331-51c2a257278c.png)

```
        sqlServer.Databases.Define("databaseName")
            .WithEdition(DatabaseEdition.GeneralPurpose)
```

I don't see any of the operators allowing me to specify serverless

**Comment by weidongxu-microsoft** [comment]
@Tyler-V 

You can uise `ServiceObjectiveName newValue = ServiceObjectiveName.Parse("GP_S_Gen5_10");` to make one.

**Comment by weidongxu-microsoft** [comment]
The lib is deprecated https://github.com/Azure/azure-libraries-for-net

Recommend migrate to https://aka.ms/azsdk/dotnet/mgmt

---

Maybe backend changed its behavior. That would be a regression on their part.

However you might really want to migrate, as their old api-version 2014 would be retired in next year.

**Comment by Tyler-V** [comment]
@weidongxu-microsoft yes some possible regression but was a good time for us to migrate, the documentation was lack-luster so ill share our solution for the next person that may find this issue

```
    private readonly ArmClient _armClient;
    private readonly AppSettings _appSettings;

    public MyService(IOptions<AppSettings> appSettings)
    {
        _appSettings = appSettings.Value;
        _armClient = new ArmClient(new DefaultAzureCredential(new DefaultAzureCredentialOptions()
        {
            TenantId = _appSettings.Azure.TenantId
        }));
    }
```

```
    private async Task AddServerlessDatabasesAsync(IEnumerable<DbContext> dbContexts)
    {
        var sqlServerResource = _armClient.GetSqlServerResource(ResourceIdentifier.Parse(_appSettings.Azure.SqlServer.ResourceId));
        var sqlDatabaseCollection = sqlServerResource.GetSqlDatabases();

        var existingDatabases = new List<string>();
        await foreach (var sqlDatabaseResource in sqlDatabaseCollection.GetAllAsync())
        {
            existingDatabases.Add(sqlDatabaseResource.Data.Name);
        }

        var databaseNames =
            dbContexts.Select(dbContext => new SqlConnectionStringBuilder(dbContext.Database.GetDbConnection()
                .ConnectionString).InitialCatalog);

        var tasks = databaseNames.Except(existingDatabases)
            .Select(databaseName => Task.Run(async () =>
            {
                try
                {
                    await sqlDatabaseCollection.CreateOrUpdateAsync(WaitUntil.Completed, databaseName, new SqlDatabaseData(AzureLocation.EastUS2)
                    {
                        AutoPauseDelay = 60,
                        MinCapacity = 0.5,
                        IsZoneRedundant = false,
                        RequestedBackupStorageRedundancy = _appSettings.Azure.SqlServer.BackupStorageRedundancy,
                        Sku = new SqlSku(_appSettings.Azure.SqlServer.Sku),
                        Tags = { { "Environment", EnvironmentHelper.EnvironmentName }, { "Service Line", "DPE" } }
                    });
                }
                catch (Exception)
                {
                    // ignored
                }
            }));

        await Task.WhenAll(tasks);
    }
```
